corehq/apps/userreports/views.py

```diff
@@ -515,14 +521,22 @@ def _get_config_kwargs(self, app_source):
             )
         }
 
-    def _build_data_source(self, app_source, username):
+    def _expire_data_source(self, data_source_config_id):
+        with temp_setting_value('CELERY_ALWAYS_EAGER', False):  # so it works in dev environments too
+            delete_data_source_task.apply_async(
+                (self.domain, data_source_config_id),
+                countdown=TEMP_DATA_SOURCE_LIFESPAN
+            )
+
+    def _build_temp_data_source(self, app_source, username):
         data_source_config = DataSourceConfiguration(
             domain=self.domain,
             table_id=_clean_table_name(self.domain, uuid.uuid4().hex),
             **self._get_config_kwargs(app_source)
         )
         data_source_config.validate()
         data_source_config.save()
+        self._expire_data_source(data_source_config._id)
```

Comment:
It isn't, but that might be a better workflow than leaving it lying around for a day.